FRCs/frc-00zz.md

```diff
@@ -0,0 +1,73 @@
+## Simple Summary
+Timelock encryption is a method of encrypting a message now that can't be decrypted until a specific point in the future has passed. The drand team have launched a scheme that uses BLS12-381 signatures in an innovative way using [identity based encryption](https://crypto.stanford.edu/~dabo/papers/bfibe.pdf) to achieve practical Timelock encryption using the drand network.
+As the Filecoin network already supports BLS12-381, it could trivially implement built-in FVM actors for enabling timelock encryption (providing the [FIP for switching to the new drand network](https://github.com/filecoin-project/FIPs/pull/652) is accepted).
+
+
+## Abstract
+On 2023-03-01, drand launched the first practical Timelock encryption scheme to drand mainnet after undergoing a security audit and developing libraries and tooling to make it easier.
+Timelock encryption would enable users to do sealed-bid auctions, create more secure implementations of [RANDAO](https://github.com/randao/randao), schedule social media network messages, prevent [MEV](https://coinmarketcap.com/alexandria/glossary/miner-extractable-value-mev), and much, much more!
+The drand timelock scheme uses [identity based encryption](https://crypto.stanford.edu/~dabo/papers/bfibe.pdf) which exploits the pairing capabilities of BLS12-381 to use a future round number as a public key to encrypt messages which will be decryptable only once that future round has been emitted by the drand network by using the round signature as a private key.
+
+We propose a precompile for FVM that would enable users to timelock encrypt arbitrary data and attempt decryption of previously timelock encrypted data. This would enable users to timelock encrypt some part of the data inside their smart contract, while also enabling users to pass ciphertexts directly into smart contracts for decryption there at a later time. 
+
+## Change Motivation
+To enable timelock encryption and decryption using primitives that are already available in the FVM internals.
+
+## Specification
+
+The following interfaces should be fulfilled in FEVM and native actors respectively:
+
+```solidity
+interface Timelock {
+    function encrypt(bytes message, uint64 roundNumber) returns (bytes)
```

Comment:
I would not expose this API which is hard to use for users - I would either use "block number" from Filecoin or even better "timestamp" and then internally the API figure out which drand number it is.
Also this method in particular can not be done "onchain" so it's a bit weird to express it as a stg that "should be fulfilled in FEVM" , it has to be done on the client side.